x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/Expressions.java

```diff
@@ -127,22 +137,51 @@ public static Pipe pipe(Expression e) {
         throw new SqlIllegalArgumentException("Cannot create pipe for {}", e);
     }
 
-    public static TypeResolution typeMustBe(Expression e, Predicate<Expression> predicate, String message) {
-        return predicate.test(e) ? TypeResolution.TYPE_RESOLVED : new TypeResolution(message);
+    public static TypeResolution typeMustBeBoolean(Expression e, String operationName, ParamOrdinal paramOrd) {
+        return typeMustBe(e, dt -> dt == DataType.BOOLEAN, operationName, paramOrd, "boolean");
+    }
+
+    public static TypeResolution typeMustBeInteger(Expression e, String operationName, ParamOrdinal paramOrd) {
+        return typeMustBe(e, dt -> dt.isInteger, operationName, paramOrd, "integer");
```

Comment:
You can use a method handle for consistency: `DataType::isInteger`

Reply:
I'd prefer to keep it as is, since `isInteger` is a field and not a function, like `isRational`. The functions have more "complex" evaluation like `isNumeric() => isInteger || isRational`.